content/blog/2022/10/envvar-policy.md

```diff
@@ -0,0 +1,146 @@
+
+```yaml
+settings:
+  rules:
+    - reject: anyNotIn
+      environmentVariables:
+        - name: "envvar2"
+          value: "envvar2_value"
+        - name: "envvar3"
+```
+
+### allAreUsed
+
+`allAreUsed` the container cannot use all listed volumes at once. Checks if all of the `environmentVariables` are in the Pod/Workload resource:
+
+```yaml
+settings:
+  rules:
+    - reject: allAreUsed
+      environmentVariables:
+        - name: "envvar2"
+          value: ""
+```
+
+### notAllAreUsed
+
+`notAllAreUsed` the container can use all listed volumes at once, but only one of them. Checks if all of the `environmentVariables` are not in the Pod/Workload resource
+
+```yaml
+settings:
+  rules:
+    - reject: notAllAreUsed
+      environmentVariables:
+        - name: "envvar3"
+          value: "envvar3_value"
+        - name: "envvar4"
+          value: "envvar4_value"
+```
+
+
+The user is allowed to define multiple rules in the policy settings. The resource must be approved by all of them in order to be accepted into the cluster.
+
+## An example
+
+Let's see how users can enforce that their resources have a minimum set of required environment variables. For that, it possible to use the `allAreUsed` operator:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: policies.kubewarden.io/v1
+kind: ClusterAdmissionPolicy
+metadata:
+  name: environment-variable-policy
+spec:
+  # the following version is not released yet. Waiting for https://github.com/kubewarden/environment-variable-policy/pull/9
+  module: ghcr.io/kubewarden/policies/environment-variable-policy:v0.1.1
+  mutating: false
+  rules:
+  - apiGroups: ["apps"]
+    apiVersions: ["v1"]
+    resources: ["deployments"]
+    operations:
+    - CREATE
+    - UPDATE
+  settings:
+    rules:
+      - reject: anyIn
+        environmentVariables:
+          - name: "foo"
+            value: "bar"
+          - name: "foo2"
+            value: "bar2"
+
+EOF
+```
+
+With the above policy only resources that have **all** the variables defined in the `environmentVariables` field will be allowed.
+
+Now try to deploy a resource that violates that rule:
+
+
+```console
+kubectl apply -f - <<EOF
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  labels:
+    app: nginx
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: template-nginx
+  template:
+    metadata:
+      labels:
+        app: template-nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.2
+        ports:
+        - containerPort: 80
+        env:
+          - name: "foo"
+            value: "bar"
+EOF
+# the following error message should be updated after https://github.com/kubewarden/environment-variable-policy/pull/9
+Error from server: error when creating "deployment.yaml": admission webhook "clusterwide-environment-variable-policy.kubewarden.admission" denied the request: Resource misses at least one environment variable from the rule.
+```
+
+
+Try it out!
```

Comment:
Now that we have multiple sections we have to create a dedicated one for the closing words.

Maybe something like "Call to action", "Summary" or something similar...

```md
## Summary

We hope this article provided you a nice overview about the potential use cases of this policy and how to use it.
Give it a try and, as usual....
```